src/App.tsx

```diff
@@ -1,14 +1,63 @@
-import React from 'react';
+import React, {
+  useCallback, useMemo, useRef, useState,
+} from 'react';
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { People } from './components/People';
+
+type Props = {
+  delay: number
+};
+
+export const App: React.FC<Props> = ({ delay }) => {
+  const [user, setUser] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [selectedUser, setSelectedUser] = useState('');
+
+  const timerId = useRef(0);
+
+  const findUser = (username: string) => {
+    return peopleFromServer.find(person => person.name === username);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setUser(event.target.value);
+    window.clearTimeout(timerId.current);
+
+    if (user.length <= 1) {
+      setSelectedUser('');
+    }
+
+    timerId.current = window.setTimeout(() => {
```

Comment:
move all "delay" set logic to a separate function with `debounce` name. You can google more about original debounce function